services/blockstorage/service.go

```diff
@@ -297,10 +333,55 @@ func (s *service) validateProtocolVersion(blockPair *protocol.BlockPairContainer
 }
 
 // TODO: this should not be called directly from CommitBlock, it should be called from a long living goroutine that continuously syncs the state storage
-func (s *service) syncBlockToStateStorage(commitedBlockPair *protocol.BlockPairContainer) error {
+func (s *service) syncBlockToStateStorage(committedBlockPair *protocol.BlockPairContainer) error {
 	_, err := s.stateStorage.CommitStateDiff(&services.CommitStateDiffInput{
-		ResultsBlockHeader: commitedBlockPair.ResultsBlock.Header,
-		ContractStateDiffs: commitedBlockPair.ResultsBlock.ContractStateDiffs,
+		ResultsBlockHeader: committedBlockPair.ResultsBlock.Header,
+		ContractStateDiffs: committedBlockPair.ResultsBlock.ContractStateDiffs,
 	})
 	return err
 }
+
+func (s *service) validateWithConsensusAlgos(prevBlockPair *protocol.BlockPairContainer, lastCommittedBlockPair *protocol.BlockPairContainer) error {
+	for _, handler := range s.consensusBlocksHandlers {
+		_, err := handler.HandleBlockConsensus(&handlers.HandleBlockConsensusInput{
+			BlockType:              protocol.BLOCK_TYPE_BLOCK_PAIR,
+			BlockPair:              lastCommittedBlockPair,
+			PrevCommittedBlockPair: prevBlockPair,
+		})
+
+		// one of the consensus algos has validated the block, this means it's a valid block
+		if err == nil {
+			return nil
+		}
+	}
+
+	return errors.New("all consensus algos refused to validate the block")
```

Comment:
Add count of them (all 3 consensus algos) to avoid issues that some of them did not register yet

Reply:
Fixed